src/geocat/comp/meteorology.py

```diff
@@ -1431,3 +1431,155 @@ def saturation_vapor_pressure_slope(
             tfill)
 
     return svp_slope
+
+
+def _calc_deltapressure_1D(pressure_lev, surface_pressure):
+    """Helper function for `calc_deltapressure1. Calculates the pressure layer
+    thickness (delta pressure) of a one-dimensional pressure level array.
+
+    Returns an array of length matching `pressure_lev`.
+
+    Parameters
+    ----------
+    pressure_lev : :class:`np.Array`
+        The pressure level array. May be in ascending or descending order.
+        Must have the same units as `surface_pressure`.
+
+    surface_pressure : :class:`float`
+        The scalar surface pressure. Must have the same units as
+        pressure_lev`.
+
+    Returns
+    -------
+    delta_pressure : :class:`np.Array`
+        The pressure layer thickness array. Shares dimensions and units of
+        `pressure_lev`.
+    """
+    pressure_top = min(pressure_lev)
+
+    # Safety checks
+    if not surface_pressure:
+        warnings.warn("'surface_pressure` can't equal a missing value.")
+    if pressure_top <= 0:
+        warnings.warn("'pressure_lev` values must all be positive.")
+    if pressure_top > surface_pressure:
+        warnings.warn(
+            "`surface_pressure` must be greater than minimum `pressure_lev` value."
+        )
+
+    # Sort so pressure increases (array goes from top of atmosphere to bottom)
+    is_pressuredecreasing = pressure_lev[1] < pressure_lev[0]
+    if is_pressuredecreasing:
+        pressure_lev = np.flip(pressure_lev)
+
+    # Calculate delta pressure
+    delta_pressure = np.empty_like(pressure_lev)
+
+    delta_pressure[0] = (pressure_lev[0] +
+                         pressure_lev[1]) / 2 - pressure_top  # top level
+    for i in (np.arange(1, len(pressure_lev) - 1)):  # middle levels
+        delta_pressure[i] = (pressure_lev[i + 1] - pressure_lev[i - 1]) / 2
+        i += 1
```

Comment:
I think this will also need some vectorization for future dask integration

Reply:
Similarly, I don't expect a user to be chunking along the one-dimensional pressure_lev, but along a dimension of surface_pressure. But I guess they always could find a reason? I don't know much about vectorization for dask integration tbh. I could use some help with that.

Reply:
You know what, that's true enough though. Let's just make sure nothing in here will force a compute. 

And we can probably re-arrange this to be a little more pythonic, though. Something like 

```suggestion
    delta_pressure[1:-1] = [(a-b)/2 for a, b in zip(pressure_lev[2:], pressure_lev[:-1])]
```